### ISSUE TITLE
:bug: apply and create git repo success, then delete from github, apply again and fail

### ISSUE THREAD

**Issue by JaneLiuL** [merico-dev/stream#263]
## Describe the Bug
below is my config, the first time i run apply -f ./config.yml
and check the repo on github, and i delete the new repo
then i run again apply -f ./config.yml , it fail as below
<img width="936" alt="image" src="https://user-images.githubusercontent.com/56822274/156546397-ecb85887-9f1d-414f-abab-d5ae997e218a.png">

```
---
tools:
# name of the instance with github-repo-scaffolding-golang
- name: golang-demo
  plugin:
    kind: github-repo-scaffolding-golang
    version: 0.2.0
  options:
    owner: janeliul
    # the repo which you'd like to create
    repo: test-demo
    # the branch of the repo you'd like to hold the code
    branch: main
    image_repo: xxxx/golang-demo
- name: golang-demo-actions
  plugin:
    kind: githubactions-golang
    version: 0.2.0
  options:
    owner: janeliul
    # the repo where you'd like to setup GitHub Actions
    repo: test-demo
    # programming language specific settings
    language:
      name: go
      # version of the language
      version: "1.17"
    # main branch of the repo (to which branch the plugin will submit the workflows)
    branch: main
    build:
      # default to true
      enable: True
      # build command, OPTIONAL, the given value below is default value
      command: "go build ./..."
    test:
      # default to false
      enable: False
      # test command, OPTIONAL, the given value below is default value
      command: "go test ./..."
      coverage:
        # default to false
        enable: False
        # go test profile subcommand, OPTIONAL, the given value below is default value
        profile: "-race -covermode=atomic"
        output: "coverage.out"
    docker:
      # docker build/push related, default to false
      enable: False
      # dockerhub image repo
      repo: golang-demo

```

## How to Reproduce

Steps to reproduce the bug:

1. create a new git repo via dtm
2. delete git repo from github
3. run again the apply -f
4. See the error 

## Expected Behavior
expect is should can create success

## Screenshots

If applicable, add screenshots to help explain the issue.

## Additional Context

Add any other helpful context about the issue here.

**Comment by daniel-hutao** [comment]
@JaneLiuL I went into your `repository list` and didn't find a project named `test-demo`. Please confirm that the project has been successfully created. 

It is also possible that the delete operation is performed twice.

**Comment by JaneLiuL** [comment]
i think you have some mistakes.
please apply first, then delete repo on github,
then apply again, will fail at 404.
this is a bug. i can reproduce.
and also when i delete `rm -rf ./devstream.state ` and apply again can success

**Comment by IronCore864** [comment]
Let me rephrase, see if I understand your process correctly or not:

1. add both repo-scaffolding and github-actions plugins in the config
2. apply
3. after a successful apply, the repo and the github actions are both created
4. delete the repo manually
5. apply again

Is this what you mean?

If yes, I can reproduce. I'm on it. Thanks!

---
Chinese:

为了确保我准确理解了你的意思， 让我重复描述一下，你看对不对：

1. 把repo-scaffolding和github-actions这两个插件都放在配置里
2. apply
3. 成功apply后，repo成功被创建、github actions也成功被创建
4. 手动删除上一步创建出来的repo
5. 再次apply，报错

我理解对了吗？如果对了的话，那我明白你的意思了，并且我也可以复现。我来修复一下。如果没理解对，还请再解释一下，谢谢，也可以用中文，没有问题，我可以负责翻译成英文给国外的社区朋友们看。

**Comment by IronCore864** [comment]
https://github.com/merico-dev/stream/blob/a7ea9480f1db36db9287619291bf80b40c57cbc1/pkg/util/github/repo.go#L44

This might be the root cause. When 404, err isn't nil.

**Comment by IronCore864** [comment]
PR #266 is for this issue.

We will keep this issue open until the next bug-fix release is made.
